models/concerns/expand_data.rb

```diff
@@ -80,14 +56,14 @@ def expand_unavailable_days
     self.vehicles.each{ |vehicle|
       vehicle.unavailable_days |= unavailable_days
     }
-    [self.services + self.shipments].flatten.each{ |mission|
+    self.services.each{ |mission|
       mission.unavailable_days |= unavailable_days
     }
   end
 
   def provide_original_info
-    (self.services + self.shipments + self.vehicles).each{ |element|
-      element.original_id = element.id
+    (self.services + self.vehicles).each{ |element|
+      element.original_id ||= element.id
```

Comment:
Failing scheduling tests come from this edit. Why should we keep original_id ? This field is not provided by users (not in grape model).

Reply:
When converting shipment into services we edit the id of the pickup and the delivery are edited, we assign the original id at this moment. Then, at this point, the original id may already be defined